easybuild/easyblocks/g/gromacs.py

```diff
@@ -498,3 +554,115 @@ def sanity_check_step(self):
             'dirs': dirs,
         }
         super(EB_GROMACS, self).sanity_check_step(custom_paths=custom_paths)
+
+    def run_all_steps(self, *args, **kwargs):
+        """
+        Put configure options in place for different variants, (no)mpi, single/double precision.
+        """
+        # Save installopts so we can reset it later. The gmxapi pip install
+        # can't handle the -j argument.
+        self.save_installopts = self.cfg['installopts']
+
+        # keep track of config/build/installopts specified in easyconfig
+        # file, so we can include them in each iteration later
+        common_config_opts = self.cfg['configopts']
+        common_build_opts = self.cfg['buildopts']
+        common_install_opts = self.cfg['installopts']
+
+        self.save_install_cmd = self.cfg.get('install_cmd')
+        self.save_build_cmd = self.cfg.get('build_cmd')
+
+        self.cfg['configopts'] = []
+        self.cfg['buildopts'] = []
+        self.cfg['installopts'] = []
+
+        if LooseVersion(self.version) < LooseVersion('4.6'):
+            prec_opts = {
+                'single': '',
+                'double': '--enable-double',
+            }
+            mpi_type_opts = {
+                'nompi': '',
+                'mpi': '--enable-mpi'
+            }
+        else:
+            prec_opts = {
+                'single': '',
+                'double': '-DGMX_DOUBLE=ON',
+            }
+            mpi_type_opts = {
+                'nompi': '-DGMX_MPI=OFF -DGMX_THREAD_MPI=ON',
+                'mpi': '-DGMX_MPI=ON -DGMX_THREAD_MPI=OFF'
+            }
+
+        if LooseVersion(self.version) < LooseVersion('5'):
+            # For older versions we only build/install the mdrun part for
+            # the MPI variant. So we need to be able to specify the
+            # install target depending on variant.
+            self.cfg['install_cmd'] = 'make'
+
+            # Use the fact that for older versions we just need to
+            # build and install mdrun for the MPI part
+            build_opts = {
+                'nompi': '',
+                'mpi': 'mdrun'
+            }
+            install_opts = {
+                'nompi': 'install',
+                'mpi': 'install-mdrun'
+            }
+        else:
+            build_opts = {
+                'nompi': '',
+                'mpi': ''
+            }
+            install_opts = {
+                'nompi': 'install',
+                'mpi': 'install'
+            }
+
+        precs = ['single']
```

Comment:
nitpick: Use `precisions`. Had to scroll back to this to see what `for prec in precs:` means